client/src/plugins/camunda-plugin/deployment-tool/DeploymentConfigModal.js

```diff
@@ -144,35 +113,60 @@ export default class DeploymentConfigModal extends React.PureComponent {
 
   }
 
+  onAuthDetection = (isAuthNeeded) => {
+    this.setState({
+      isAuthNeeded
+    });
+  }
+
+  checkAuthStatusOnlyOnce = (values) => {
+    if (this.shouldCheckIfAuthNeeded) {
+      this.props.validator.validateConnectionWithoutCredentials(values.endpoint.url).then((result) => {
+        if (!result) {
+          this.onAuthDetection(false);
+        } else if (!result.isExpired) {
+          this.onAuthDetection(!!result && (result.code === 'UNAUTHORIZED'));
+        }
+      });
+      this.shouldCheckIfAuthNeeded = false;
+    }
+  }
+
   render() {
 
     const {
       fieldError,
       onSubmit,
-      validate,
-      onClose
+      onClose,
+      onAuthDetection
     } = this;
 
     const {
       configuration: values,
       validator,
       title,
       intro,
-      primaryAction
+      primaryAction,
+      saveCredential,
+      removeCredentials
     } = this.props;
 
     const {
-      connectionState
+      isAuthNeeded
     } = this.state;
 
+    // @oguz:
+    // FormIK validateOnMount get executed only once but not everytime
+    // DeploymentConfigModal is mounted. Thats why we need this logic here.
+    this.checkAuthStatusOnlyOnce(values);
```

Comment:
I think we could simply check this via `componentDidMount`. Cf. https://github.com/camunda/camunda-modeler/pull/1721/commits/b7ecdc7fe682b4943acb8576f4ad4078cae099b2 with attached test case. What do you think?

Reply:
Ah yeah, this is actually a really good idea! 🍰